PHPCompatibility/Sniffs/PHP/DeprecatedMagicAutoloadSniff.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * \PHPCompatibility\Sniffs\PHP\DeprecatedMagicAutoloadSniff.
+ *
+ * PHP version 7.2
+ *
+ * @category  PHP
+ * @package   PHPCompatibility
+ * @author    Wim Godden <wim.godden@cu.be>
+ */
+
+namespace PHPCompatibility\Sniffs\PHP;
+
+use PHPCompatibility\Sniff;
+
+/**
+ * \PHPCompatibility\Sniffs\PHP\DeprecatedMagicAutoloadSniff.
+ *
+ * @category  PHP
+ * @package   PHPCompatibility
+ * @author    Wim Godden <wim.godden@cu.be>
+ */
+class DeprecatedMagicAutoloadSniff extends Sniff
+{
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return array(T_FUNCTION);
+    }//end register()
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer_File $phpcsFile The file being scanned.
+     * @param int                   $stackPtr  The position of the current token in the
+     *                                         stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(\PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+    {
+        if ($this->supportsAbove('7.2') === false) {
+            return;
+        }
+
+        $funcName = $phpcsFile->getDeclarationName($stackPtr);
+        
+        if (strtolower($funcName) !== '__autoload') {
+            return;
+        }
+
+        if ($this->determineNamespace($phpcsFile, $stackPtr) != '') {
+            return;
+        }
+
+        if ($this->validDirectScope($phpcsFile, $stackPtr, array('T_CLASS', 'T_ANON_CLASS', 'T_INTERFACE', 'T_TRAIT')) === true) {
```

Comment:
Ah! Found it. The `ValidDirectScope()` method expects the string tokens as keys in the array, so it can use the more efficient `isset()`.

So it expects the array to be passed like this:
```php
$excludeScopes = array(
	'T_CLASS'      => true,
	'T_ANON_CLASS' => true,
	'T_INTERFACE'  => true,
	'T_TRAIT'      => true,
	'T_NAMESPACE'  => true, // = Only scoped namespaces, non-scoped still needs to be checked in another way.
);

if ($this->validDirectScope($phpcsFile, $stackPtr, $excludedScopes) === true) {
	....
```

As `$excludedScopes` does not changes between calls to the sniff, it should probably be declared as a private property of the sniff.